src/circuit.rs

```diff
@@ -92,6 +92,64 @@ pub struct Cell {
     column: Column<Any>,
 }
 
+/// An allocated cell.
+#[derive(Clone, Copy, Debug)]
+pub struct AllocatedCell<F: FieldExt, T: Copy + Into<F>> {
+    value: Option<T>,
+    cell: Cell,
+    _marker: PhantomData<F>,
+}
+
+impl<F: FieldExt, T: Copy + Into<F>> AllocatedCell<F, T> {
+    /// Instantiate a new AllocatedCell.
+    pub fn new(value: Option<T>, cell: Cell) -> Self {
```

Comment:
Should this be `pub` or `pub(crate)`? The `halo2-gadgets` crate will likely need to use it.